atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -277,8 +277,17 @@ def run(self):
 
         task_id, filesystem_regex = self.build_filesystem(image_build_conf)
 
-        task = TaskWatcher(self.session, task_id, self.poll_interval)
-        task.wait()
+        try:
+            task = TaskWatcher(self.session, task_id, self.poll_interval)
+            task.wait()
+        except BuildCanceledException as exc:
+            self.log.info("Build was canceled, canceling task %s", task_id)
+            try:
+                self.session.cancelTaskFull(task_id)
```

Comment:
Ah, correct, `cancelTask` should be sufficient, I was confused by 'getTaskChildren' mocking in tests. Fixed